similar/main/fireball.cpp

```diff
@@ -1380,25 +1380,31 @@ unsigned do_exploding_wall_frame(wall &w1)
 
 		auto &Walls = LevelUniqueWallSubsystemState.Walls;
 		auto &vmwallptr = Walls.vmptr;
-		auto &w2 = *vmwallptr(csegp->shared_segment::sides[cside].wall_num);
-		assert(&w1 != &w2);
-		assert((w1.flags & WALL_EXPLODING) || (w2.flags & WALL_EXPLODING));
-		w1.flags |= WALL_BLASTED;
-		w2.flags |= WALL_BLASTED;
-		if (w1_explode_time_elapsed >= EXPL_WALL_TIME)
+
+		auto cwall_num = csegp->shared_segment::sides[cside].wall_num;
+		if (cwall_num != wall_none)
```

Comment:
Now that C++17 is used, you can use an init-statement in a selection statement.  I have not made an effort to make this conversion in old code, though.